include/emds/emds.h

```diff
@@ -112,9 +113,20 @@ int emds_entry_add(struct emds_dynamic_entry *entry);
  *
  * Triggers the process of storing all data registered to be stored. All data
  * registered either through @ref emds_entry_add function or the
- * @ref EMDS_STATIC_ENTRY_DEFINE macro is stored. Once the data storage is
- * completed, the data should not be changed, and the device should be halted.
+ * @ref EMDS_STATIC_ENTRY_DEFINE macro is stored. It locks all interrupts until
+ * the write is finished. Once the data storage is completed, the data should
+ * not be changed, and the device should be halted. The device must not be
+ * allowed to reboot when operating on a backup supply. Since reboot will
+ * trigger data load from EMDS storage and clear the storage area. The reboot
+ * should be allowed only when the main power supply is available.
+ *
  * This is a time-critical operation and will be processed as fast as possible.
+ * To achieve better time predictability, this function must be called from an
+ * interrupt context with the highest priority.
+ *
+ * When writing to flash it bypasses the flash driver. Therefore, when running
+ * with MPSL, make sure to uninitialize MPSL before calling this function.
+ * Otherwise, an assertion may be triggered by the exit of the function.
```

Comment:
```suggestion
 * When writing to flash, it bypasses the flash driver. Therefore, when running
 * with MPSL, make sure to uninitialize the MPSL before this function is called.
 * Otherwise, an assertion may be triggered by the exit of the function.
```